File: typescript-playwright/src/api/CategoryApi.ts
import type { APIRequestContext, APIResponse } from "@playwright/test";
import { API_URL } from "../utils/constants";

export class CategoryApi {
  private readonly request: APIRequestContext;
  private readonly authHeaders: Record<string, string> = {};
  constructor(
    request:APIRequestContext,
    authHeaders: Record<string, string>,
  ) {
    this.request = request;
    this.authHeaders = authHeaders;
  }

  async getAll(): Promise<APIResponse> {
    return this.request.get(`${API_URL}/api/categories`);
  }

  async create(name: string): Promise<APIResponse> {
    return this.request.post(`${API_URL}/api/categories`, {
      data: { name },
      headers: this.authHeaders,
    });
  }

  async deleteCategory(
    categoryId: number,
    headers?: Record<string, string>,
  ): Promise<APIResponse> {
    return this.request.delete(`${API_URL}/api/categories/${categoryId}`, {
      headers: headers ?? this.authHeaders,
    });
  }
}

File: python-pytest/src/api/category_api.py
from __future__ import annotations

import requests

from src.utils.constants import API_URL


class CategoryApi:
    def __init__(self, session: requests.Session, auth_headers: dict[str, str]) -> None:
        self._session = session
        self._auth_headers = auth_headers

    def get_all(self) -> requests.Response:
        return self._session.get(f"{API_URL}/api/categories", timeout=10)

    def create(self, name: str) -> requests.Response:
        return self._session.post(
            f"{API_URL}/api/categories",
            json={"name": name},
            headers=self._auth_headers,
            timeout=10,
        )

    def delete_category(
        self, category_id: int, headers: dict[str, str] | None = None
    ) -> requests.Response:
        return self._session.delete(
            f"{API_URL}/api/categories/{category_id}",
            headers=headers or self._auth_headers,
            timeout=10,
        )
